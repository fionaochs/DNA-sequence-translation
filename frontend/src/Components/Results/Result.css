h3 {
    font-size: 1.5rem;

}

.sequenceResult {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
}
