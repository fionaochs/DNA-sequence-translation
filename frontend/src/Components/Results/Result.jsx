import React from 'react';
import styles from './Result.css';

const Result = ({DNASequence, proteinName, organism, proteinLocation}) => {

    const sequenceNotFound = proteinLocation === undefined;

    return (
        <div className={styles.sequenceResult}>
            {sequenceNotFound ? (
                <div>
                    <h3>Sequence '{DNASequence}'</h3>
                    <h3>did not match any organisms on file</h3>
                </div>
            ) : (
                <div>
                    <h3 className={DNASequence}>Found sequence</h3>
                    <h3>'{DNASequence}'</h3>
                    <h3>with amino acid code '{proteinName}'</h3>
                    <h3>in organism '{organism}'</h3>
                    <h3>at location {proteinLocation}</h3>
                </div>
            )
            }
        </div>
    );
};

export default Result;