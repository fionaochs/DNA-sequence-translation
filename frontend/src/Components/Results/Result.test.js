import React from 'react';
import {render, screen, fireEvent, waitFor, cleanup} from '@testing-library/react';
import Result from './Result';

describe('Rating component', () => {
    afterEach(() => cleanup());
    it('renders Result', () => {
        let result = {};
        const {asFragment} = render(<Result result={result}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('changes the result form when new result submitted', async () => {
        // const {container} = render(<Result />);
        // const DNASequence = container.querySelector('DNASequence')
        const DNASequence = await screen.getByDisplayValue('DNASequence');

        // fireEvent.change(container.firstChild.value, {target: {value: 'AACGTTA'}});
        // await waitFor(() => {
        //     expect(container.firstChild).toBe('AACGTTA');
        // });
        fireEvent.change(DNASequence, {target: {value: 'AACGTTA'}});
        await waitFor(() => {
            expect(DNASequence.value).toBe('AACGTTA');
        });
    });
    it('find result text in display', async () => {
        render(<Result />);
        const linkElement = screen.getByText(/Found sequence/i);
        expect(linkElement).toBeInTheDocument();
    });
});