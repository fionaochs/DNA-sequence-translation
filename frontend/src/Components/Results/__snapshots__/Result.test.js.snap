// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component renders Result 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Found sequence
    </h3>
    <h3>
      ''
    </h3>
    <h3>
      with amino acid code ''
    </h3>
    <h3>
      in organism ''
    </h3>
    <h3>
      at location 
    </h3>
  </div>
</DocumentFragment>
`;
